core/app/models/spree/taxon/paperclip_attachment.rb

```diff
@@ -22,4 +22,13 @@ def icon_present?
   def attachment_partial_name
     'paperclip'
   end
+
+  def destroy_attachment(definition)
+    return false unless respond_to?(definition)
+
+    attached_file = send(definition)
+    return false unless attached_file
+
+    attached_file.destroy
+  end
```

Comment:
I think it may be worth writing a few model specs for this method. The integration test is great, but of course it covers only the happy path.

Reply:
added!